commit 553a282cb25eb62fdda1f3425d48b12372366e03
Author: Johannes Berg <johannes.berg@intel.com>
Date:   Sun Jul 17 18:16:51 2022 +0200

    wifi: mac80211: mlme: fix override calculation
    
    In my previous changes here, I neglected to take the old
    conn_flags into account that might still be present from
    the authentication, and thus ieee80211_setup_assoc_link()
    can misbehave, as well as the override calculation being
    wrong. Fix that by ORing in the old flags.
    
    Fixes: 1845c1d4a455 ("wifi: mac80211: mlme: refactor assoc link setup")
    Signed-off-by: Johannes Berg <johannes.berg@intel.com>

diff --git a/net/mac80211/mlme.c b/net/mac80211/mlme.c
index ee1a72519c15..2d389e74c953 100644
--- a/net/mac80211/mlme.c
+++ b/net/mac80211/mlme.c
@@ -7048,6 +7048,8 @@ int ieee80211_mgd_assoc(struct ieee80211_sub_if_data *sdata,
 		goto err_clear;
 	}
 
+	/* keep old conn_flags from ieee80211_prep_channel() from auth */
+	conn_flags |= link->u.mgd.conn_flags;
 	conn_flags |= ieee80211_setup_assoc_link(sdata, assoc_data, req,
 						 conn_flags, assoc_link_id);
 	override = link->u.mgd.conn_flags != conn_flags;